lib/schema/schema.js

```diff
@@ -24,8 +24,14 @@ var util = require('util');
 
 var misc = require('../misc.js');
 
+exports.validDeviceTime = function(val) {
+  if (!/^(\d{4}-[01]\d-[0-3]\dT[0-2]\d:[0-5]\d:[0-5]\d)$/.test(val)) {
```

Comment:
I'm willing to revert the validTimestamp change below, but you commented on the deviceTime change.

Date.parse() will not prevent improperly-formatted deviceTimes, which cannot have a timezone offset (they are local display time).

We could do a combination of both for deviceTime, or...?